Updates to meter api and schema

jongoncalves opened the pull request:
- I updated the enum values for `MeterType` - you'll have to nuke your DB and re-run migrations
- Updated ARW to handle single instances being returned vs a list
- I added a service location schema so the data gets auto serialized
@avpeery if the response needs to be flat and not nested, we can just change the Meter schema to flatten it out and not have to change the API. Let me know if it needs to be flat.
- Moved ARW and Schema instantiation to function scope. This prevents it from being instantiated when the file gets loaded.

On the get single meter api, `channels` seems to be missing. Is that right?
Also, I haven't added the interval stuff to the serialization just yet. But let me know if this is making sense and I can merge it into your branch and then branch out for the rest.

avpeery commented:
Thank Jonathan - I am checking out the changes right now!

avpeery commented:
Hey Jonathan so I am pulled this branch, nuked the database, and am putting some sample data in to run the api routes, and noted that channels (while it is models, for some reason wasn't pick up by the database migration). I am looking into it.

jongoncalves commented:
> Hey @jongoncalves so I am pulled this branch, nuked the database, and am putting some sample data in to run the api routes, and noted that channels (while it is in models, for some reason wasn't pick up by the database migration). I am looking into it.
> 
> Ah - just needed to run flask db migrate, and it picked up channels

Nice!

avpeery commented:
Hi @jongoncalves, received info that json object should be flattened

jongoncalves commented:
> Hi @jongoncalves, received info that json object should be flattened

ok, good to know, that should be easy enough to reflect on the MeterSchema

jongoncalves commented on web/web/api/v1/meter.py:
```diff
@@ -65,80 +61,68 @@ def show_meter_info(meter_id):
+    # 'rate': meter.get_rates(),
+    # 'interval_count': meter.get_interval_count(interval_count_start, interval_count_end), 
+    # 'interval_coverage': Interval.get_interval_coverage(interval_coverage), 
+    # 'exports': 'NOT YET CREATED'
```
Left these here so we add them in due time...

jongoncalves commented on web/web/api/v1/meter.py:
```diff
@@ -65,80 +61,68 @@ def show_meter_info(meter_id):
+        return arw.to_json()
 
-    results = meter_schema.dump(modified_meter, many=True)
+    results = meter_schema.dump(modified_meter)
```
@avpeery  check out the latest working put route here

jongoncalves commented on web/web/api/v1/meter_api_schema.py:
```diff
@@ -1,96 +1,100 @@
+            'example': ''
+        }
+    }
+]
```
autopep8 re-formatted the file - nothing else

jongoncalves commented on web/web/api/v1/response_wrapper.py:
```diff
@@ -44,6 +44,9 @@ def __init__(self):
         }
     
+    def has_errors(self):
+        return len(self.response["errors"]) != 0
```
convenience method

jongoncalves commented on web/web/models/meter.py:
```diff
@@ -96,10 +98,31 @@ def __repr__(self):
         load_instance = True
-        include_fk = True
\ No newline at end of file
+        include_fk = True
```
@avpeery checkout the updated schema here. We should try to overload the schema with the responsibility to create the ingest format and output responses as much as possible. Also, this should be where we consolidate all the validation per model, since we should try to use the schema as much as possible to marshal data back and forth.

avpeery replied:
Hey thanks so much for your work on this - I totally see where you are going, and it makes a lot more sense to me now. This looks good to me

jongoncalves commented:
- Fixed the double serialization issue due to `dump` vs `dumps` on the GET endpoint
- Added the fields query string param to the GET endpoint so we can actually retrieve multiple meter models (not just ids) at the same time. To retrieve just the ids, add: `?fields=meter_id`
- Removed the meter `backref` from the Interval model
- Removed duplicated code from the MeterSchema WRT meter_type
- Update the POST endpoint so it actually only creates instead of create or update. Will research this more in the future.

@avpeery Check out the latest in this PR this morning and let me know if the GET, PUT and POST work as expected for you. They are all finally working for me. Let's get this merged in this morning if we can. 👍

avpeery commented:
Ok I am running through the routes now! Both get routes work! 

Came across an issue with PUT, I think it's expected that the json object on the PUT route shouldn't expect any input for map location since that is a column on the service location table, error I received 'AttributeError: 'NoneType' object has no attribute 'map_location''.. I have not done any debugging yet. POST route does not work as expected - I think you commented on this. It doesn't throw error for primary key duplication like it should/was.

jongoncalves commented:
> Ok I am running through the routes now! Both get routes work!
> 
> Came across an issue with PUT, I think it's expected that the json object on the PUT route shouldn't expect any input for map location since that is a column on the service location table, error I received 'AttributeError: 'NoneType' object has no attribute 'map_location''.. I have not done any debugging yet. POST route does not work as expected - I think you commented on this. It doesn't throw error for primary key duplication like it should/was.
> 
> -post route is now picking up the error! Same code, maybe just didn't register the changes the first flask run?, but it POST route works now
> -put route also works! good to go!

Thanks, will merge this in to your branch and then master, is that ok?

avpeery commented:
Yes that's great thanks!
